packages/utils/package-manager.ts

```diff
@@ -66,6 +26,30 @@ export function getPackageManager(): string {
 	}
 }
 
+/**
+ *
+ * Spawns a new process using the respective package manager
+ *
+ * @param {String} pkg - The dependency to be installed
+ * @param {Boolean} isNew - indicates if it needs to be updated or installed
+ * @returns {Function} spawn - Installs the package
+ */
+
+function spawnWithArg(pkg: string, isNew: boolean): SpawnSyncReturns<Buffer> {
+	const packageManager: string = getPackageManager();
```

Comment:
@ematipico It seems that the `getPackageManager()` method has some ambiguous logic which breaks those tests :thinking:

Reply:
https://github.com/webpack/webpack-cli/blob/6170fd0197b0aea8e00d3dc397ed754bc043d299/packages/utils/package-manager.ts#L15

Reply:
You can mock the same way, I agree. You need to revise nonetheless

Reply:
@evenstensberg How to proceed further?

Reply:
@evenstensberg The following test script for detecting yarn as the package manager fails each and every time.
https://github.com/webpack/webpack-cli/blob/6170fd0197b0aea8e00d3dc397ed754bc043d299/packages/utils/__tests__/package-manager.test.ts#L38

Reply:
@ematipico Thoughts :thinking:

Reply:
@evenstensberg The following test script for detecting yarn as the package manager fails each and every time.
https://github.com/webpack/webpack-cli/blob/6170fd0197b0aea8e00d3dc397ed754bc043d299/packages/utils/__tests__/package-manager.test.ts#L38

Reply:
@misterdev Thoughts :thinking: